### PR TITLE
VM: Implement NZ MUL opcode

### LINKED ISSUE #223: VM: implement NZ , MUL opcode

### PR DESCRIPTION
**Pull request by dauTT**
Closes #223

### CODE REVIEW

File: pkg/vm/vm_ops_maths.go [hunk 1]
```diff
@@ -94,6 +94,45 @@ func Dec(op stack.Instruction, ctx *stack.Context, istack *stack.Invocation, rst
 	return NONE, nil
 }
 
+// Nz returns a stack Item whose value is:
```
Comment (decentralisedkev) [inline pkg/vm/vm_ops_maths.go]: Nz pops an integer from the stack.
Then pushes a boolean to the stack which evaluates to true
iff the integer was zero
Returns an error if the popped item cannot be casted to an integer
or if we cannot create a boolean

Comment (dauTT) [reply pkg/vm/vm_ops_maths.go]: > iff the integer was zero

you mean:

iff the integer was **not** zero

correct?

Comment (decentralisedkev) [reply pkg/vm/vm_ops_maths.go]: Yep sorry, `evaluates to true iff the integer was non-zero`

Comment (dauTT) [reply pkg/vm/vm_ops_maths.go]: :+1:

**Comment by hal0x2328** [comment]
100 points dauTT

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/630c74684cc6dad330397c63e8c7aa6403c043cc1f241638cff87e3b97a40716

**Pull request merged**